packages/ffe-account-selector-react/src/components/account-selector/AccountSelector.js

```diff
@@ -1,152 +1,189 @@
+            `${value}`
+                .replace(/\s/g, '') // Remove all spaces
+                .replace(/\./g, '') // Remove all periods
+                .toLowerCase();
+        const cleanedSearchString = cleanString(searchString);
+        return searchAttributes
+            .map(searchAttribute => cleanString(item[searchAttribute]))
+            .some(cleanedItemAttribute =>
+                cleanedItemAttribute.includes(cleanedSearchString),
+            );
+    };
 
-    assignBaseSelectorRef(baseSelector) {
-        this.baseSelector = baseSelector;
-    }
+    const onInputChange = event => {
+        setInputValue(event.target.value);
+        if (inputProps && inputProps.onChange) {
+            inputProps.onChange();
+        }
+    };
 
-    renderNoMatches() {
-        return (
-            <AccountNoMatch
-                value={this.props.noMatches}
-                locale={this.props.locale}
-            />
-        );
-    }
+    const handleAccountSelected = value => {
+        const hasResetSelection = value === null;
+        const hasSelectedCustomAccount = !value?.accountNumber;
+        if (hasResetSelection) {
+            setInputValue('');
+            onReset();
+        } else if (hasSelectedCustomAccount) {
+            onAccountSelected({ name: '', accountNumber: value.name });
+        } else {
+            onAccountSelected(value);
+        }
+    };
 
-    onAccountSelect(account) {
-        this.enableFilter = false;
-        this.props.onAccountSelected(account);
-    }
+    const formatter = formatAccountNumber
+        ? formatIncompleteAccountNumber
+        : undefined;
 
-    onInputChange(value) {
-        this.enableFilter = true;
-        this.props.onChange(value);
-    }
+    const customNoMatch = allowCustomAccount
+        ? {
+              dropdownList: [
+                  {
+                      name: formatter ? formatter(inputValue) : inputValue,
+                      accountNumber: '',
+                  },
+              ],
+          }
+        : noMatches;
 
-    onSuggestionSelect(suggestion) {
-        if (suggestion) {
-            this.baseSelector.showOrHideSuggestions(false, () =>
-                this.onAccountSelect(suggestion),
-            );
-        }
-    }
+    const dropdownAttributes = showBalance
+        ? ['name', 'accountNumber', 'balance']
+        : ['name', 'accountNumber'];
 
-    filterSuggestions() {
-        const { value, accounts } = this.props;
-        const suggFilt = createAccountFilter(this.enableFilter);
-        return accounts.filter(suggFilt(value));
-    }
-
-    render() {
-        const {
-            className,
-            id,
-            locale,
-            selectedAccount,
-            showBalance,
-            highCapacity,
-        } = this.props;
-        return (
+    return (
+        //Allows styling based on saveSpaceForDetails even when AccountSelector is wrapped in InputGroup
+        <div className="disable-margin-collapse">
```

Comment:
Denne er litt hacky, jeg tar med glede imot forslag til forbedringer :pray:

Reply:
i første omgang bør klassnavnet forholde seg til BEM iaf. Vad er det vi ønsker uppnå?

Reply:
Det er for å kunne holde av plass til accountDetails selvom man har `InputGroup` med `extraMargin={false}`. `InputGroup` sin margin har høyere spesifisitet enn vår egen (klassisk InputGroup-problem..), så den overstyrer marginen vi prøver å bruke for å spare plass til accountDetails.
![image](https://user-images.githubusercontent.com/3830142/117271069-21a9d500-ae5a-11eb-964f-0da0f6b037ad.png)
Det er stjålet fra denne https://stackoverflow.com/questions/19718634/how-to-disable-margin-collapsing#answer-33132624

Men jeg er enig i at det er veldig funky og ganske hacky, så hvis du har noen forslag til bedre løsning er jeg lutter øre :smile:

Reply:
Etter prat med @kwltrs endra jeg den til `ffe-account-selector-container` uten å prefikse alle andre klassenavn med det :+1: